### PR TITLE
feat: Add ingestion status for References

### PR DESCRIPTION
**Pull request by gjreda**
This is still a WIP for now. Need to fix one bug and add/update tests.
_____
part of #113

This PR updates ingest to:
1. Copy `uploads` and work from a `.staging` directory so that each new upload does not process previously ingested References
2. Adds a `status` field to `Reference` which displays the "completed" ingestion status
3. Be a bit cleaner and easier to read

### CODE REVIEW

File: python/sidecar/ingest.py [hunk 1]
```diff
@@ -26,90 +32,210 @@
...
+        jstore.load()
+        return jstore.references
 
-    def get_files_to_ingest(self) -> list[Path]:
+    def _get_files_to_ingest(self) -> list[Path]:
         """
-        Determines which files need to be ingested
-        :return: bool
+        Determines which files need to be ingested by comparing the list of 
+        already processed References against files found in `uploads`.
```
Comment (cguedes) [inline python/sidecar/ingest.py:137]: Wouldn't be easier to give you the list of file paths to ingest? In the FE the user selects files and not directories.

File: python/cli.schema.json [hunk 2]
```diff
@@ -82,6 +82,9 @@
         "citation_key": {
           "type": "string"
         },
+        "status": {
+          "type": "string"
```
Comment (cguedes) [inline python/cli.schema.json:86]: Can we have a enum here (i.e. expected strings for status)?

File: python/sidecar/ingest.py [hunk 3]
```diff
@@ -26,90 +32,210 @@
 handler.setFormatter(formatter)
 
 logger.addHandler(handler)
-logger.disabled = os.environ.get("SIDECAR_ENABLE_LOGGING", "false").lower() == "true"
+logger.disabled = os.environ.get("SIDECAR_DISABLE_LOGGING", "true").lower() == "true"
```
Comment (cguedes) [inline python/sidecar/ingest.py:35]: Do you want to make logging active by default?

**Comment by gjreda** [comment]
Since we merged #181 for fetching ingest status, I am going to close this PR and break out the pieces of it that are related to #218 has a new PR.

**Pull request closed without merging**